MapboxCoreNavigation/RouteController.swift

```diff
@@ -839,8 +835,7 @@ extension RouteController: CLLocationManagerDelegate {
         let lastKnownUserAbsoluteDistance = routeProgress.currentLegProgress.currentStepProgress.userDistanceToManeuverLocation
 
         if userSnapToStepDistanceFromManeuver <= RouteControllerManeuverZoneRadius &&
-            courseMatchesManeuverFinalHeading ||
-            (userAbsoluteDistance > lastKnownUserAbsoluteDistance && lastKnownUserAbsoluteDistance > RouteControllerManeuverZoneRadius) {
+            (courseMatchesManeuverFinalHeading || (userAbsoluteDistance > lastKnownUserAbsoluteDistance && lastKnownUserAbsoluteDistance > RouteControllerManeuverZoneRadius)) {
```

Comment:
Take a look at what the code was before: https://github.com/mapbox/mapbox-navigation-ios/pull/1013/files#diff-1a2149692b3d7c58b6a0392120d6011aL836

With the change in https://github.com/mapbox/mapbox-navigation-ios/pull/1013/files#diff-1a2149692b3d7c58b6a0392120d6011aL836, we we're advance the step unnecessarily because it would evaluate to true too often.